geodepy/heights_s3.py

```diff
@@ -0,0 +1,234 @@
+#___________________________________________________________________________#
+# Some notes:
+# Written by Jack McCubbine of Geoscience Australia, date: 08/11/2019
+# This code contains functions to handle tranformations between GPS and 
+# AWVS/AHD and Vice Versa 
+# Gridded data used for the varisous reference surfaces are geotiff files
+# These allow direct access remotely using "gdal"
+# File names are hard coded 
+# Some exmaples of how the functions can be used are give towards the end 
+#___________________________________________________________________________#
+# Import dependencies 
+import gdal
+import numpy as np
+from scipy.interpolate import griddata
+#___________________________________________________________________________#
+# Interpolation functions
+def interp_DOVPV(Lat,Long):
+    # Import the DOVPM file
+    f = gdal.Open('/vsicurl/https://geoid.s3-ap-southeast-2.amazonaws.com/AVWS/DOV_PV.tif')
+    # load band (akin to a variable in dataset)
+    band = f.GetRasterBand(1)   
+    # get the pixel width, height, etc.
+    transform = f.GetGeoTransform()
+    # Grid resolution (known)
+    res=transform[1]
+    # convert lat,lon to row,col
+    column = (Long - transform[0]) / transform[1]
+    row = (Lat - transform[3]) / transform[5]
+    # get pixel values surrounding data point
+    Surrounding_data=(band.ReadAsArray(np.floor(column-2), np.floor(row-2), 5, 5))
+    # convert row,col back to north,east
+    Long_c = transform[0] + np.floor(column) * res
+    Lat_c = transform[3] - np.floor(row) * res
+    # set up matrices for interpolation
+    count=-1
+    pos=np.zeros((25,2))
+    Surrounding_data_v=np.zeros((25,1))
+    for k in range(-2,3):
+        for j in range(-2,3):
+            count=count+1
+            pos[count]=(Long_c+j*res,Lat_c-k*res)
+            Surrounding_data_v[count]=Surrounding_data[k+2,j+2]          
+    interp_DOVPV_val=griddata(pos,Surrounding_data_v,(Long,Lat),method='cubic')
+    return interp_DOVPV_val
+
+def interp_DOVPM(Lat,Long):

```

Comment:
Would it be possible to get docstrings for these functions explaining their inputs, outputs and associated units?